hs_core/views/resource_rest_api.py

```diff
@@ -896,6 +908,10 @@ def post(self, request, pk):
     def delete(self, request, pk, filename):
         resource, _, user = view_utils.authorize(
             request, pk, needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+        view_utils.irods_path_is_allowed(filename)
```

Comment:
This is a systematic problem for the whole of resource_rest_api.py. This will return 404 because an exception is thrown. This should be 400: bad request. Fixed my code, but didn't fix the entirety of resource_rest_api.py.....